### PR TITLE
fix auth problems, ensure routes are protected

### PR DESCRIPTION
**Pull request by sdkalchofner**
Have to use some not so nice code to check if the user is logged in. Depending if one does a hard reload or not, it takes a second for the data from store to be loaded, in which case often its detected as one is not logged in, but actually is. This new solution prevents this with using a req to the api as fallback.

### CODE REVIEW

File: frontend/src/routes/PrivateRouteGuard.svelte [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  import { API } from "../ts/api";
+
+  import { user, error } from "../ts/mainStore";
+
+  let userFromStoreOrAPI = {};
+  onMount(async () => {
+    // have to do this to prevent wrong behaviour from loading it solely via store...
+    try {
+      userFromStoreOrAPI =
+        typeof $user.id == "undefined" ? await API.user.get() : $user;
```
Comment (andyundso) [inline frontend/src/routes/PrivateRouteGuard.svelte:13]:
```suggestion
        $user.id == undefined ? await API.user.get() : $user;
```

**Review by andyundso — COMMENTED** [review]
wondering: since you removed parts of d515412638a5afcafe0975ebb162289c425935ba with this PR, could more code be removed or those the remaining code handle different stuff than what you introduced?

**Comment by sdkalchofner** [comment]
@andyundso > wondering: since you removed parts of https://github.com/flatfeestack/flatfeestack/commit/d515412638a5afcafe0975ebb162289c425935ba with this PR, could more code be removed or those the remaining code handle different stuff than what you introduced? 

Honestly, didn't even look into that commit, really. Not exactly sure what the whole users' page etc, should be for. But I don't see a need for that anymore. I'll try to clean up accordingly.

**Review by andyundso — APPROVED** [review]
nice! :meat_on_bone:

**Pull request merged**